[2024-10-05T00:00:01.732+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-10-05T00:00:01.748+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_job_dag.submit_spark_job scheduled__2024-10-04T00:00:00+00:00 [queued]>
[2024-10-05T00:00:01.758+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_job_dag.submit_spark_job scheduled__2024-10-04T00:00:00+00:00 [queued]>
[2024-10-05T00:00:01.759+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 2
[2024-10-05T00:00:01.774+0000] {taskinstance.py:2888} INFO - Executing <Task(SparkSubmitOperator): submit_spark_job> on 2024-10-04 00:00:00+00:00
[2024-10-05T00:00:01.779+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=5423) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-10-05T00:00:01.780+0000] {standard_task_runner.py:72} INFO - Started process 5424 to run task
[2024-10-05T00:00:01.781+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'spark_job_dag', 'submit_spark_job', 'scheduled__2024-10-04T00:00:00+00:00', '--job-id', '1115', '--raw', '--subdir', 'DAGS_FOLDER/jobsSpark.py', '--cfg-path', '/tmp/tmpip88hnfo']
[2024-10-05T00:00:01.782+0000] {standard_task_runner.py:105} INFO - Job 1115: Subtask submit_spark_job
[2024-10-05T00:00:01.828+0000] {task_command.py:467} INFO - Running <TaskInstance: spark_job_dag.submit_spark_job scheduled__2024-10-04T00:00:00+00:00 [running]> on host f8603ea05473
[2024-10-05T00:00:01.915+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_job_dag' AIRFLOW_CTX_TASK_ID='submit_spark_job' AIRFLOW_CTX_EXECUTION_DATE='2024-10-04T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-10-04T00:00:00+00:00'
[2024-10-05T00:00:01.916+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-10-05T00:00:01.942+0000] {base.py:84} INFO - Retrieving connection 'spark_default'
[2024-10-05T00:00:01.945+0000] {spark_submit.py:403} INFO - Spark-Submit cmd: spark-submit --master spark://spark:7077 --num-executors 2 --executor-memory 1G --driver-memory 512M --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py
[2024-10-05T00:00:06.170+0000] {spark_submit.py:579} INFO - usage: WebScraping_Transformation.py [-h] --json_file_path JSON_FILE_PATH
[2024-10-05T00:00:06.172+0000] {spark_submit.py:579} INFO - --path_save_transfor PATH_SAVE_TRANSFOR
[2024-10-05T00:00:06.173+0000] {spark_submit.py:579} INFO - WebScraping_Transformation.py: error: the following arguments are required: --json_file_path, --path_save_transfor
[2024-10-05T00:00:06.243+0000] {spark_submit.py:579} INFO - 24/10/05 00:00:06 INFO ShutdownHookManager: Shutdown hook called
[2024-10-05T00:00:06.246+0000] {spark_submit.py:579} INFO - 24/10/05 00:00:06 INFO ShutdownHookManager: Deleting directory /tmp/spark-2ca10d35-a9e7-4028-918e-aa25b0bcac3a
[2024-10-05T00:00:06.286+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 505, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --num-executors 2 --executor-memory 1G --driver-memory 512M --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py. Error code is: 2.
[2024-10-05T00:00:06.299+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=spark_job_dag, task_id=submit_spark_job, run_id=scheduled__2024-10-04T00:00:00+00:00, execution_date=20241004T000000, start_date=20241005T000001, end_date=20241005T000006
[2024-10-05T00:00:06.319+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-10-05T00:00:06.320+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 1115 for task submit_spark_job (Cannot execute: spark-submit --master spark://spark:7077 --num-executors 2 --executor-memory 1G --driver-memory 512M --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py. Error code is: 2.; 5424)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 505, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --num-executors 2 --executor-memory 1G --driver-memory 512M --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py. Error code is: 2.
[2024-10-05T00:00:06.340+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-10-05T00:00:06.357+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-10-05T00:00:06.362+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
